website/docs/concepts/router.md

```diff
@@ -198,16 +207,161 @@ additional functionality that is not available in `AnyHistory` and
+
+    let go_home_button = {
+        let onclick = Callback::once({
+            let history = history.clone();
+            move |_| history.push(Route::Home)
+        });
+        html!{<button {onclick}>{"click to go home"}</button>}
+    };
+
+    let go_to_first_post_button = {
+        let onclick = Callback::once({
+            let history = history.clone();
+            let onclick = Callback::once(move |_| history.push(Route::Post{id: "first-post".to_string()}));
+        });
+        
+        html!{<button {onclick}>{"click to go the first post"}</button>}
+    };
+
+    let go_to_secure_button = {
+        let onclick = Callback::once(move |_| history.push(Route::Secure));
+        html!{<button {onclick}>{"click to go to the last post"}</button>}
+    };
+
+    html! {
+        <>
+            {go_home_button}
+            {go_to_first_post_button}
+            {go_to_secure_button}
+        </>
+    }
+}
+```
+
+##### normal components
+
+For normal components, the `AnyHistory` instance can be obtained through the `ctx.link().history()` API. The 
+rest is identical with the functional component case. Here's an example of a view function that renders a single button.
+
+```rust ,ignore
+fn view(&self, ctx: &Context<Self>) -> Html {
+    let history = ctx.link().history().unwrap();
+    let onclick = Callback::once(move |_| history.push(MainRoute::Home));
+    html!{<button {onclick}>{"Go Home"}</button>}
+}
+```
+
+#### Redirect
+
+`yew-router` also provides a `<Redirect/>` element in the prelude. 
+It can be used to achieve similar effects as the history API. The element accepts a
+`to` attribute as the target route.
+When a `<Redirect/>` element is rendered, it internally calls `history.push()` and changes the route.
+Here is an example:
+
+```rust
+#[function_component(SomePage)]
+fn some_page() -> Html {
+    // made-up hook `use_user`
+    let user = match use_user() {
+        Some(user) => user,
+        // an early return that redirects to the login page
+        // technicality: `Redirect` actually renders an empty html. But since it also pushes history, the target page
+        // shows up immediately. Consider it a "side-effect" component.
+        None => return html! { <Redirect to={Route::Login}/> },
+    };
+    // ... actual page content.
+}
+```
+:::tip `Redirect` vs `history`, which to use 
+The history API is the only way to manipulate route in callbacks. 
+While `<Redirect/>` can be used for early returns in a component.
+You might also want to use `<Redirect/>` in other
+non-component context, for example in the switch function of a [Nested Router](#nested-router).
+:::
+
 
 ### Listening to Changes
 
 #### Functional components
 
-Simply use available hooks `use_history`, `use_location` and `use_route`.
+Alongside the `use_history` hook, there's also `use_location` and `use_route`.
 Your components will re-render when provided values change.
 
-#### Struct components
+#### Normal components
```

Comment:
They do, but `struct` you can associate that you had defined a struct for that component.
I am really against calling them `normal` because a function component is normal for me aswell. Its a buzzword

Reply:
yes I agree with you. "struct component" is a nice name. I'll change